pkg/vsphere/tasks/waiter.go

```diff
@@ -27,12 +28,40 @@ import (
 	"github.com/vmware/govmomi/vim25/progress"
 	"github.com/vmware/govmomi/vim25/soap"
 	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/pkg/trace"
 )
 
 const (
 	maxBackoffFactor = int64(16)
+
+	VMContextObjectKey = "VMContextObject"
+)
+
+var (
+	m             sync.RWMutex
+	errorHandlers []func(ctx context.Context, err error) (bool, error)
```

Comment:
Define a type for the function type, e.g. `type ErrorHandler func(ctx context.Context, err error) (bool, error)`